sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_encode.py

```diff
@@ -765,14 +765,15 @@ def encode_filter_set(value):
         else:
             if isinstance(name, str):
                 name = name.encode("utf-8")  # type: ignore
-            try:
+            if isinstance(data, (str, bytes)):
+                described_filter = data
+            # handle the situation when data is a tuple or list of length 2
```

Comment:
do we know if it's always be str/bytes/tuple/list? is there any way for it to be another type? would we want to still keep the try except inside the else?

Reply:
good catch, there should be a try/except inside there in case something else comes through. This is also far deep in the pyamqp stack that it would be us adding in some new filter versus an end user. Something we need to think about while typing this function for MQ

Reply:
ah yep yep that makes sense! just for the sake of testing, maybe we could throw int/long/dict etc. values to see how it would handle them.